### PR TITLE
Fixes for fluidsynth >= 2.0.0

### PR DESCRIPTION
**Pull request by schnitzeltony**
Build tested with fluidsynth 1.1.11 and 2.0.1.

### CODE REVIEW

File: source/Makefile.mk [hunk 1]
```diff
@@ -221,7 +221,7 @@ endif
 endif
 
 HAVE_FFMPEG     = $(shell pkg-config --exists libavcodec libavformat libavutil && echo true)
-HAVE_FLUIDSYNTH = $(shell pkg-config --atleast-version=1.1.7 fluidsynth && pkg-config --max-version=1.9.9 fluidsynth && echo true)
+HAVE_FLUIDSYNTH = $(shell pkg-config --atleast-version=1.1.7 fluidsynth && echo true)
```
Comment (falkTX) [inline source/Makefile.mk]: Looks great, thanks very much!
Sorry my github is somehow screwed up, does not allow me to comment on the conversation tab without closing the issue, so I will make comments here.

Please make this target develop branch, not master. all new things should now go into develop first, then cherry-picked from there
There are some other small things, I will write them on the relevant lines.

File: source/backend/plugin/CarlaPluginFluidSynth.cpp [hunk 2]
```diff
@@ -73,20 +73,22 @@ class CarlaPluginFluidSynth : public CarlaPlugin
         fSynth = new_fluid_synth(fSettings);
         CARLA_SAFE_ASSERT_RETURN(fSynth != nullptr,);
 
+        keepFluidDefaults();
```
Comment (falkTX) [inline source/backend/plugin/CarlaPluginFluidSynth.cpp]: I dont like some the naming here. please rename to `InitializeFluidDefaultsIfNeeded` (which I believe is a lot more clear on what it does.

Comment (falkTX) [reply source/backend/plugin/CarlaPluginFluidSynth.cpp]: err, lowercase I, so `initializeFluidDefaultsIfNeeded`

File: source/backend/plugin/CarlaPluginFluidSynth.cpp [hunk 3]
```diff
@@ -1683,13 +1768,19 @@ class CarlaPluginFluidSynth : public CarlaPlugin
     float** fAudio16Buffers;
     float   fParamBuffers[FluidSynthParametersMax];
 
+    static bool  bFluidDefaultsKept;
```
Comment (falkTX) [inline source/backend/plugin/CarlaPluginFluidSynth.cpp]: These class static variables should have prefix "s" (both of them)

**Comment by schnitzeltony** [comment]
Story continues with V2  at https://github.com/falkTX/Carla/pull/766

**Pull request closed without merging**